src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -180,10 +180,15 @@ private void checkExists(String path) {
 
   private void connect() throws JSchException {
     session.connect();
-    Channel channel = session.openChannel(CHANNEL_SFTP);
-    channel.connect();
-
-    sftp = (ChannelSftp) channel;
+    try {
+      Channel channel = session.openChannel(CHANNEL_SFTP);
+      channel.connect();
+      sftp = (ChannelSftp) channel;
+    } catch (JSchException ex) {
+      LOGGER.debug("There was an error when opening the channel", ex.getCause());
```

Comment:
That cause is never logged because a missing placeholder in the format string.
Add the placeholder and include the throwable as third argument.

Reply:
My Bad, we usually log as follows
  LOGGER.debug("There was an error when opening the channel {}", ex.getMessage(),ex);
or you can just leave as
  LOGGER.debug("There was an error when opening the channel", ex);

Reply:
Solved!